qbraid_qir/qasm3/visitor.py

```diff
@@ -801,6 +803,35 @@ def _validate_variable_assignment_value(self, variable: Variable, value: Any) ->
         else:
             raise TypeError(f"Invalid type {type_to_match} for variable {variable.name}")
 
+    def _validate_array_assignment_values(
+        self, variable: Variable, dimensions: List[int], values: List[Any]
```

Comment:
```suggestion
        self, variable: Variable, dimensions: list[int], values: list[Any]
```

As of Python 3.9 ([PEP 585](https://www.python.org/dev/peps/pep-0585/)) onwards `tuple`, `list`, `dict`, and various other classes are now generic types. Using these rather than their `typing` counterpart is preferred to minimize the runtime impact of typing.

- https://stackoverflow.com/a/39458225/14781553
- https://docs.python.org/3/whatsnew/3.9.html#type-hinting-generics-in-standard-collections

Reply:
Okay got it, I'll tackle that in the new issue.